[2025-08-18T16:25:46.940+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: churn_pipeline_dag.data_validation manual__2025-08-18T16:25:39.604409+00:00 [queued]>
[2025-08-18T16:25:46.950+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: churn_pipeline_dag.data_validation manual__2025-08-18T16:25:39.604409+00:00 [queued]>
[2025-08-18T16:25:46.951+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-08-18T16:25:46.961+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): data_validation> on 2025-08-18 16:25:39.604409+00:00
[2025-08-18T16:25:46.969+0000] {standard_task_runner.py:57} INFO - Started process 84 to run task
[2025-08-18T16:25:46.972+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'churn_pipeline_dag', 'data_validation', 'manual__2025-08-18T16:25:39.604409+00:00', '--job-id', '46', '--raw', '--subdir', 'DAGS_FOLDER/churn_pipeline_dag.py', '--cfg-path', '/tmp/tmpaa4nx98f']
[2025-08-18T16:25:46.973+0000] {standard_task_runner.py:85} INFO - Job 46: Subtask data_validation
[2025-08-18T16:25:47.024+0000] {task_command.py:416} INFO - Running <TaskInstance: churn_pipeline_dag.data_validation manual__2025-08-18T16:25:39.604409+00:00 [running]> on host dc6694cfc37c
[2025-08-18T16:25:47.104+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='churn_pipeline_dag' AIRFLOW_CTX_TASK_ID='data_validation' AIRFLOW_CTX_EXECUTION_DATE='2025-08-18T16:25:39.604409+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-18T16:25:39.604409+00:00'
[2025-08-18T16:25:48.010+0000] {python.py:194} INFO - Done. Returned value was: Starting data validation...
data_lake/raw/source=web_logs/ingestion_date=2025-08-18/run_id=20250818_162544/part-00001.csv passed all checks
data_lake/raw/source=web_logs/ingestion_date=2025-08-17/run_id=20250817_053855/part-00001.csv passed all checks
data_lake/raw/source=web_logs/ingestion_date=2025-08-17/run_id=20250817_053536/part-00001.csv passed all checks
Data quality report saved to reports/data_quality_report.csv

[2025-08-18T16:25:48.040+0000] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=churn_pipeline_dag, task_id=data_validation, execution_date=20250818T162539, start_date=20250818T162546, end_date=20250818T162548
[2025-08-18T16:25:48.069+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-08-18T16:25:48.095+0000] {taskinstance.py:2778} INFO - 1 downstream tasks scheduled from follow-on schedule check
